libiop/bcs/common_bcs_parameters.tcc

```diff
@@ -15,10 +15,11 @@ bcs_transformation_parameters<FieldT, MT_root_hash> default_bcs_params(
     params.hash_enum = hash_type;
     /* TODO: Push setting leaf hash into internal BCS code. Currently 2 is fine, as leaf size is internally unused. */
     const size_t leaf_size = 2;
-    params.leafhasher_ = get_leafhash<FieldT, MT_root_hash>(hash_type, security_parameter, leaf_size);
+    params.cap_size = 2;
+    params.leafhasher_ = get_leafhash<MT_root_hash, FieldT>(hash_type, security_parameter, leaf_size);
     params.compression_hasher = get_two_to_one_hash<MT_root_hash, FieldT>(hash_type, security_parameter);
-    params.hashchain_ =
-        get_hashchain<FieldT, MT_root_hash>(hash_type, security_parameter);
+    params.cap_hasher = get_cap_hash<MT_root_hash, FieldT>(hash_type, security_parameter);
+    params.hashchain_ = get_hashchain<FieldT, MT_root_hash>(hash_type, security_parameter);
```

Comment:
As discussed above, we are switching `<FieldT, MT_root_hash>` to `<MT_root_hash, FieldT>` in `leafhasher` (now `cap_hasher` here). 

Do we want to apply the same refactoring to `hashchain`? We probably should do that in a separate PR for ease of testing.

Reply:
Yeah I'd prefer to keep it the same for this PR

Reply:
If it's acceptable, I'd like to keep them both the current order for now. We can refactor both at the same time in a future PR.